### PR TITLE
Make `profiles` and `update_profiles` custom fields non-searchable for avoiding errors with too large DB indices

### LINKED ISSUE #47: Error updating to 1.1-beta due to exceeding key length for DB indices of custom fields
https://github.com/systopia/de.systopia.remoteevent/blob/f392e21fde10527172bfc39b3278de61a65896b7/CRM/Remoteevent/Upgrader.php#L179 tries to alter the `text_length` of a custom field to 1024 which causes an error during extension DB updates:

```
ALTER TABLE civicrm_value_remote_registration 
        MODIFY `profiles` varchar(1024), 
        MODIFY `update_profiles` varchar(1024) [nativecode=1071 ** Specified key was too long; max key
 length is 3072 bytes]
```
Ping to @dontub who added this change.

Maybe making the field not searchable would be a workaround, as their columns will probably not cause an index be created, but this will have to be done in the `resources/custom_group_remote_registration.json`, as this is being synced within the upgrade task.

### PR DESCRIPTION
**Pull request by jensschuppe**
Fixes #47 (as a workaround, that is).

Increasing those fields' `VARCHAR` length to 1024 might cause `[nativecode=1071 ** Specified key was too long; max key
 length is 3072 bytes]` errors when updating to version 0.5.0. Making them non-searchable does not cause indices be created (resp. them being deleted).

Not sure this needs thorough reviews, but feel free to leave your thoughts, @dontub and @bjendres!

### CODE REVIEW

**Review by bjendres — CHANGES_REQUESTED** [review]
Makes sense to me.

But shouldn't we add an upgrader step, too? A simple
```
        $customData = new CRM_Remoteevent_CustomData(E::LONG_NAME);
        $customData->syncCustomGroup(E::path('resources/custom_group_remote_registration.json'));
```
should be enough

Although, I don't know if removing the flag drops the index...would have to test that.

**Comment by jensschuppe** [comment]
I'll look into that again. Since the upgrade step that causes the issue will not have been run, there might be no additional upgrade step needed.

**Comment by dontub** [comment]
The error does not happen with all installations of MariaDB. Thus I think an additional upgrade step makes sense.

**Comment by jensschuppe** [comment]
Might not hurt, but if the error occurred previously, a new version with just the JSON file changed will make the failed upgrade step work when attempting to re-run it. Best solution would be replacing the JSON file with managed entities though.

File: CRM/Remoteevent/Upgrader.php [hunk 1]
```diff
@@ -193,6 +193,22 @@ public function upgrade_0015()
...
+    /**
+     * Synchronizes custom fields for CiviRemote Event registration
+     * configuration fields, as two fields need to be marked as not searchable
+     * for avoiding exceeding index lengths with bigger VARCHAR lengths
+     * introduced with upgrade_0015().
+     *
+     * @return void
+     * @throws \Exception
+     */
+    public function upgrade_0016()
```
Comment (dontub) [inline CRM/Remoteevent/Upgrader.php:205]: I'd suggest to use PHP type hint instead of PHPDoc type for the return type.

**Review by bjendres — APPROVED** [review]
looks good to me

File: CRM/Remoteevent/Upgrader.php (hunk 1 above)
Comment (jensschuppe) [reply CRM/Remoteevent/Upgrader.php]: Done. Also, this made me see the missing `return true` for the upgrader, so thanks for nitpicking :wink:

**Pull request merged**